pkg/minikube/machine/start.go

```diff
@@ -202,6 +206,30 @@ func timedCreateHost(h *host.Host, api libmachine.API, t time.Duration) error {
 	}
 }
 
+// postStartValidations are validations against the host after it is created
+func postStartValidations(h *host.Host, drvName string) {
+	if !driver.IsKIC(drvName) {
+		return
+	}
+	// make sure /var isn't full, otherwise warn
+	output, err := h.RunSSHCommand("df -h /var | awk 'NR==2{print $5}'")
+	if err != nil {
+		glog.Warningf("error running df -h /var: %v", err)
+	}
+	output = strings.Trim(output, "\n")
+	percentageFull, err := strconv.Atoi(output[:len(output)-1])
+	if err != nil {
+		glog.Warningf("error getting percentage of /var that is free: %v", err)
+	}
+	if percentageFull >= 99 {
+		exit.WithError("", fmt.Errorf("docker daemon out of memory. No space left on device"))
+	}
+
+	if percentageFull > 80 {
+		out.WarningT("The docker daemon is almost out of memory, run 'docker system prune' to free up space")
```

Comment:
The advice we give to the user should be based the following

// for windows and mac docker driver users
if  driver.IsDockerDesktop(drvName) // 

Docker Desktop >. Preferences > Resources  > Increase Disk Image


if they are VM driver and they are using Docker Runtime
then they should do
- minikube ssh docker system prune
- upgrade their system's disks size or resize partitions

Or if they are VM driver and Not Docker Runtime they should or None Driver
- upgrade their system's disks size or resize partitions

Reply:
Right now I'm only handling the case for the docker driver (first couple lines of this check)

Reply:
Added the minikube ssh tip in, the output should be in the PR description